misc/bazel/c_deps/rust-sys/BUILD.rocksdb.bazel

```diff
@@ -0,0 +1,186 @@
+
+load("@rules_foreign_cc//foreign_cc:defs.bzl", "cmake")
+
+load("@aspect_bazel_lib//lib:copy_to_directory.bzl", "copy_to_directory")
+load("@aspect_bazel_lib//lib:copy_file.bzl", "copy_file")
+load("@bazel_skylib//rules:select_file.bzl", "select_file")
+
+"""Additive BUILD file for the librocksdb-sys Rust crate."""
+
+cc_library(
+    name = "snappy",
+    srcs = [
+        "snappy/snappy.cc",
+        "snappy/snappy.h",
+        "snappy/snappy-internal.h",
+        "snappy/snappy-sinksource.cc",
+        "snappy/snappy-sinksource.h",
+        "snappy/snappy-stubs-internal.cc",
+        "snappy/snappy-stubs-internal.h",
+        ":config_h",
+        ":snappy_stubs_public_h",
+    ],
+    hdrs = [
+        "snappy/snappy.h"
+    ],
+    includes = ["snappy/."],
+    copts = [
+        "-DHAVE_CONFIG_H",
+        "-fno-exceptions",
+        "-Wno-sign-compare",
+        "-Wno-shift-negative-value",
+        "-Wno-implicit-function-declaration",
+    ],
+    defines = ["HAVE_SYS_UIO_H"],
+)
+
+genrule(
+    name = "config_h",
+    outs = ["snappy/config.h"],
+    cmd = "\n".join([
+        "cat <<'EOF' >$@",
+        "#define HAVE_STDDEF_H 1",
+        "#define HAVE_STDINT_H 1",
+        "",
+        "#ifdef __has_builtin",
+        "#  if !defined(HAVE_BUILTIN_EXPECT) && __has_builtin(__builtin_expect)",
+        "#    define HAVE_BUILTIN_EXPECT 1",
+        "#  endif",
+        "#  if !defined(HAVE_BUILTIN_CTZ) && __has_builtin(__builtin_ctzll)",
+        "#    define HAVE_BUILTIN_CTZ 1",
+        "#  endif",
+        "#elif defined(__GNUC__) && (__GNUC__ > 3 || __GNUC__ == 3 && __GNUC_MINOR__ >= 4)",
+        "#  ifndef HAVE_BUILTIN_EXPECT",
+        "#    define HAVE_BUILTIN_EXPECT 1",
+        "#  endif",
+        "#  ifndef HAVE_BUILTIN_CTZ",
+        "#    define HAVE_BUILTIN_CTZ 1",
+        "#  endif",
+        "#endif",
+        "",
+        "#ifdef __has_include",
+        "#  if !defined(HAVE_BYTESWAP_H) && __has_include(<byteswap.h>)",
+        "#    define HAVE_BYTESWAP_H 1",
+        "#  endif",
+        "#  if !defined(HAVE_UNISTD_H) && __has_include(<unistd.h>)",
+        "#    define HAVE_UNISTD_H 1",
+        "#  endif",
+        "#  if !defined(HAVE_SYS_ENDIAN_H) && __has_include(<sys/endian.h>)",
+        "#    define HAVE_SYS_ENDIAN_H 1",
+        "#  endif",
+        "#  if !defined(HAVE_SYS_MMAN_H) && __has_include(<sys/mman.h>)",
+        "#    define HAVE_SYS_MMAN_H 1",
+        "#  endif",
+        "#  if !defined(HAVE_SYS_UIO_H) && __has_include(<sys/uio.h>)",
+        "#    define HAVE_SYS_UIO_H 1",
+        "#  endif",
+        "#endif",
+        "",
+        "#ifndef SNAPPY_IS_BIG_ENDIAN",
+        "#  ifdef __s390x__",
+        "#    define SNAPPY_IS_BIG_ENDIAN 1",
+        "#  elif defined(__BYTE_ORDER__) && defined(__ORDER_BIG_ENDIAN__) && __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__",
+        "#    define SNAPPY_IS_BIG_ENDIAN 1",
+        "#  endif",
+        "#endif",
+        "EOF",
```

Comment:
did you have to write this?

Reply:
I did not, copied it from tensorflow, but I just realized I forgot to add the license!

Added in https://github.com/MaterializeInc/materialize/pull/27045